### PR TITLE
Fb troubleshooter 39638

### PR DESCRIPTION
**Pull request by labkey-sweta**
Automation test

PS : Section of test is commented which verifies troubleshooter cannot save changes in authentication until the issue is fixed by rosaline.

### CODE REVIEW

**Comment by labkey-adam** [comment]
Rosaline committed her auth save fix to platform yesterday and I verified it. I just merged develop into platform/fb_troubleshooter_39638, so should be able to uncomment that test.

File: src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        log("Verify External Redirect Hosts changes cannot be saved");
+        goToAdminConsole().goToSettingsSection().clickExternalRedirectHosts();
+        checker().verifyFalse("Save should not be present for troubleshooter under External Redirect Hosts",
+                isElementPresent(Locator.button("Save")));
+
+//        log("Verify authentication changes cannot be saved");
+//        goToAdminConsole().goToSettingsSection().clickAuthentication();
+//        checker().verifyFalse("Save should not be present for troubleshooter under authentication",
+//                isElementPresent(Locator.button("Save and Finish")));
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java]: Validating the absence of something can be dangerous. If the button text for any of these page changes, these checks would pass regardless of troubleshooter access.
Put these `verifyFalse` checks in a separate method (`verifySiteSettingsPermissions(boolean canSave)`) and call it once as a site admin to verify that we're checking the correct buttons, then once as troubleshooter to verify that they aren't present.

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
Looks like you branched off of a different feature branch instead of `develop`. Karl's assay test changes shouldn't be in here.

File: src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        _userHelper.deleteUsers(false, TROUBLESHOOTER);
+    }
+
+    private void doSetup()
+    {
+        _userHelper.createUser(TROUBLESHOOTER);
+        ApiPermissionsHelper apiPermissionsHelper = new ApiPermissionsHelper(this);
+        goToSitePermissions();
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java]: Shouldn't need to navigate if you're using the API helper

File: src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+@Category({DailyC.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 4)
+public class TroubleshooterRoleTest extends BaseWebDriverTest
+{
+    private static final String TROUBLESHOOTER = "user@troubleshooter.test";
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/core/security/TroubleshooterRoleTest.java]: Avoid display name collision
```suggestion
    private static final String TROUBLESHOOTER = "troubleshooter@troubleshooter.test";
```

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**